### ISSUE TITLE
[Accessibility] Narrator does not read the status message as row added after click on Add New button of BindingNavigator

### ISSUE THREAD

**Issue by Amy-Li03** [dotnet/winforms#3596]
* .NET Core Version: 
.NET SDK 5.0.100-preview.8.20363.2

* Have you experienced this same bug with .NET Framework?: 
Yes

**More info:**
NVDA tool also can reproduce this issue.

**Minimal repro:**
1. Extract the test app, then build and run it. 
[CoreApp.zip](https://github.com/dotnet/winforms/files/4936229/CoreApp.zip)
2. Open Narrator tool, navigate to the **Add New** button of BindingNavigator.
3. Click on **Add New** button (click by mouse or through keyboard), then a row will be added.
![repro](https://user-images.githubusercontent.com/62929087/87757469-45ede400-c83d-11ea-9dfb-41ac2774c7a8.gif)

**Actual behavior:**
1. Narrator does not provide instruction that Add New button refers to  what control to be added.
2. While clicking on Add New button through Caps + Enter key, Narrator will not focus on the new added row, so Narrator does not read the status message as row added.
![Narrator1](https://user-images.githubusercontent.com/62929087/87757481-4a1a0180-c83d-11ea-8bf0-1e843b62ed79.png)

3. While clicking on Add New button by mouse, Narrator will focus on the new added row, so Narrator will read the status message as row added.
![Narrator2](https://user-images.githubusercontent.com/62929087/87757497-52723c80-c83d-11ea-84ee-c365f608e090.png)

**Expected behavior:**
1. Narrator should provide information that Add new button refers to what control should be added.
2. While clicking on Add New button through Caps + Enter key, Narrator should read the status message as row added after click on "Add New" button.

**Comment by merriemcgaw** [comment]
You may be able to do this directly in the sample app, though if we can get it into the DGV it would be pretty cool.

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @Amy-Li03. I have investigated this issue and found that this is behavior by design. After clicking on ToolStripButton, ToolStrip return focus to previous control that had focus. If user uses keyboard navigation then focus will return on ToolStrip control. As workaround i suggest to add handler with `dataGridView.Focus()` code on "Click" event of "Add new" button:
```
bindingNavigatorAddNewItem.Click += (s, e) =>
{
    dataGridView1.Focus();
};
```
This code returns focus to dataGridView.

In score of #3560 we fixed issue with supporting of AccessibleName for ToolStrip items. As result user can use AccessibleName or ToolStripText properties for providing required information to Narrator (for example: what control should be added)

**Comment by Amy-Li03** [comment]
@SergeySmirnov-Akvelon tried this workaround mentioned above, Narrator can read the status message as row added when clicking on Add New button through **Caps + Enter** key.

**Comment by SergeySmirnov-Akvelon** [comment]
This behavior is by design and the suggested workaround works for @Amy-Li03. Thus it closed as solved.
